optuna/integration/chainermn.py

```diff
@@ -41,14 +40,14 @@ class _ChainerMNObjectiveFunc(object):
             index.html#communicators>`_.
     """
 
-    def __init__(self, func, comm):
-        # type: (Callable[[ChainerMNTrial, CommunicatorBase], float], CommunicatorBase) -> None
+    def __init__(
+        self, func: "Callable[[ChainerMNTrial, CommunicatorBase], float]", comm: "CommunicatorBase"
+    ) -> None:
 
         self.comm = comm
         self.objective = func
 
-    def __call__(self, trial):
-        # type: (Trial) -> float
+    def __call__(self, trial: "Trial") -> float:
```

Comment:
I think the use of `Trial` does not cause loop reference.
```suggestion
    def __call__(self, trial: Trial) -> float:
```